### ISSUE TITLE
collect feedback on TOC and TAG engagement and employer recognition

### ISSUE THREAD

**Issue by rochaporto** [cncf/toc#1285]
Following the discussion during the TOC meeting of April 2nd, this issue tries to collect feedback on how TOC and TAG members are managing to get recognition for their engagement with the different CNCF bodies from their employers, with some context on what the employer's position is - vendor, end user, ...

Both successes and challenges are of interest. The outcome could be a document with suggestions for existing and potential members of the TOC/TAGs to help making their contributions sustainable.

**Comment by krook** [comment]
Speaking here not as an employee of the CNCF, but instead as a former member of the staff engineer review board at my previous employer that reviewed CNCF and LF contributions from promotion candidates:

As a vendor company, a person's role in open source was recognized as a key part of a career ladder that looked for broadening scope, impact, and influence in three key areas to evaluate them for promotion: Project Success; Skills Development; and Giveback (including IP and Eminence Building).

* Being an open source project contributor or maintainer had a direct impact on Project Success if the vendor built a profitable or strategic product upon it. This was recognized through quarterly awards from the OSPO (or equivalent) and promotion opportunities through the staff engineer board. This also could be used to show clear public evidence of Skills Development (hard and soft) for the board and other yearly assessments.

* Being part of a TAG or TOC would be recognized in a similar way, but more in the area of Eminence and Leadership which are key towards building up influence without authority as a respected expert in the industry. The lengthier and more substantial contributions here obviously had greater impact on a promotion case than a short term involvement that started only recently (for example, perhaps seen as an easy way to beef up their promotion package). Any leadership roles (chair, tech lead) also deepened the value of the candidate's work.

* Related to both of those types of contributions, any speaking engagements were considered valuable indicators of staff engineer growth, as these put the candidate in a position of representing the company and their work (and indirectly, their products) to conference attendees (potential customers and folks who they could recruit into their company as employees).

In short, contributions to open source and serving in a leadership role at a foundation were mutually beneficial for both the company and the growth of the individual and provide opportunities to demonstrate quantifiable impact in the industry. To support that, my former employer had incentives at quarterly, annual review, and promotion milestones.

**Comment by mrbobbytables** [comment]
@krook I don't disagree with it being a part of people's career growth, but vendors rarely TRULY value spending time in an area unless they can relate it to something more tangible. The influence is not something that can be expressed well to those who aren't familiar with the space. A person's manager may know, but as you go up the ladder, the more "concrete" something needs to be. Otherwise, the intangible stuff will be the first to go regarding a person's assigned job duties when the vendor is resource-constrained.

The TOC has firm codified responsibilities and influence. They are the ones who vote and approve projects entering and going through the levels. They also get time on the keynote stage, etc. 

TAGs (right now) have no "true" responsibility; they have some very vague influence over projects graduating levels but nothing codified. I know there are discussions around this, so it might be fixed in the future, but I do see it as something that has been an ongoing issue and may become worse if more vendors are strapped for headcount =/

The other thing that I think makes it difficult is the lack of a report or something that DOES convey the work being done in the TAGs. They get maintainer sessions, yes, but that's hard to summarize as something that could be given to a manager or further up the chain to summarize the work being done. Folks hate filling them out, but an annual or at least some kind of report that summarizes the work being done can be a huge boon. The K8s annual reports have helped multiple leads "justify" their time working on the project. I think some things can be done to make this easier for folks to complete, so it's a bit less like pulling teeth, but they do genuinely help.

**Comment by angellk** [comment]
Might be a good topic for the TAG Reboot workshop at Maintainer Summit on March 31, 2025.
cc: @rochaporto @mfahlandt

**Comment by angellk** [comment]
@TheFoxAtWork One initiative has been writing letters to employers for recognition. Perhaps create an initiative around this.

**Comment by kevin-wangzefeng** [comment]
Some notes of the discussion from today's Public TOC meeting:
1. Value of "Letters of Recognition": Personalized letters detailing a contributor's specific impact have proven valuable for their career development and in demonstrating business value to their employers.
2. Need for a Formal Process: To ensure fairness and consistency, this initiative needs to be a structured, regular process rather than an ad-hoc effort.
3. Exploring Broader Forms of Recognition: We should think beyond just letters and consider other ways to acknowledge and support our contributors. 

Possible action items to follow up:
1. Formalize the Letter Program: Document a clear, repeatable process for identifying contributors and generating these letters on a regular schedule.
2. Collect Broader Feedback: Actively solicit ideas from the community (e.g., via the TAG leads channel) on other effective forms of recognition.
3. Create Supporting Materials (template): Develop resources, e.g. a presentation deck, that contributors can use to showcase the value of their CNCF work to their employers.
